### ISSUE TITLE
Rebuild Marketo In Open Opportunity Process

### ISSUE THREAD

**Issue by nicholassewitz** [artsy/clouds#97]
Marketo's IOO process is not working. Will probably need to move to Salesforce either via APEX or a process builder.

**Comment by cfinsness** [comment]
@nicholassewitz can i see some examples of it not working?

**Comment by nicholassewitz** [comment]
yes definitely. As they come up i will put them in here. I have seen it happen twice. I looked in marketo and it shows them as being part of an opp but didn't have the action listed in the activity history.

**Comment by cfinsness** [comment]
@nicholassewitz up until now the trigger on the campaign https://app-ab14.marketo.com/#SC3145B2 was just _opportunity is updated_ and so may fail when GP _creates_ the opp at the same time as closing the opp, which happens sometimes. Just added the additional trigger _was added to opp_ w/ stage close won to see if that can capture some of those. We can test this out more thoroughly and if it doesn't work then do an alternative approach.

**Comment by cfinsness** [comment]
oh oops was looking at the Won stage instead with my last comment, but same issue may apply to the In Open Opp trigger (added versus updated).

**Comment by nicholassewitz** [comment]
@cfinsness here is an example of someone who was added to an opp but who's lifecycle wasn't changed

https://artsy.my.salesforce.com/003C0000024Wh5pIAC

**Comment by nicholassewitz** [comment]
same here https://artsy.my.salesforce.com/003C000002DnEYNIA3

**Comment by nicholassewitz** [comment]
I wonder if we should make this a batch job @cfinsness

**Comment by nicholassewitz** [comment]
For now I have created this batch job, https://app-ab14.marketo.com/#SC3917D4. We can talk about possibly putting in Salesforce if we want something more immediate

**Comment by cfinsness** [comment]
Thanks @nicholassewitz . Interesting that some of these opps were still open yet the account was closed/won since. e.g.
https://artsy.my.salesforce.com/003C0000024XkZn?srPos=0&srKp=003
https://artsy.my.salesforce.com/003C000001oglyI?srPos=0&srKp=003
